### PR TITLE
Add more detail to README

### PR DESCRIPTION
**Pull request by anne-decusatis**
Update examples to be a bit more detailed and have comments explaining what they do, clarify what is/isn't included in the derivations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,42 +82,43 @@ import magnolify.guava.semiauto._
...
 
 ```scala
+// an annotated example with Avro GenericRecord
 import magnolify.avro._
-val t = AvroType[MyCaseClass]
-val r = t.to(MyCaseClass(...))
-val c = t.from(t)
+import org.apache.avro.generic.GenericRecord
+val typeclass = AvroType[MyCaseClass] // AvroType typeclass instance defines to and from conversions between Case Class and GenericRecord
+val genericRecord: GenericRecord = typeclass.to(MyCaseClass(...)) // instantiate a case class, then pass in
```
Comment (danielblazevski) [inline README.md]: Like the expanded docs here, helped me understand this part more. 

I think it wouldn't hurt to make it more explicit, e.g. define 
```scala
case class MyCaseClass(foo: String, bar: Int)=
```
and then pass in actual args here 
```scals
val genericRecord: GenericRecord = typeclass.to(MyCaseClass("hello", 1)) // instantiate a case class, then pass in
```

File: README.md [hunk 2]
```diff
@@ -54,42 +82,43 @@ import magnolify.guava.semiauto._
...
 
-```scala
+// The other three converters work similarly to Avro: 
+
+// an example with BigQuery TableRow
 import magnolify.bigquery._
-val t = TableRowType[MyCaseClass]
-val r = t.to(MyCaseClass(...))
-val c = t.from(t)
+val typeclass = TableRowType[MyCaseClass]
```
Comment (nevillelyh) [inline README.md]: It's using `t` in some places & `typeclass` here. Make them consistent? Doesn't matter which way.

I was being lazy and using `t`, `r`, `c` for type class, record (Avro, BQ, etc.) & case class.

File: README.md [hunk 3]
```diff
@@ -54,42 +82,43 @@ import magnolify.guava.semiauto._
...
 val t = EntityType[MyCaseClass]
 val r = t.to(MyCaseClass(...))
-val c = t.from(t)
+val c = t.from(r)
 
 implicit val uriField = EntityField.from[String](URI.create)(_.toString) // custom field type
-```
 
-```scala
+// an example with TensorFlow TFExample
```
Comment (nevillelyh) [inline README.md]:
```suggestion
// an example with TensorFlow Example
```

File: README.md (hunk 2 above)
Comment (anne-decusatis) [reply README.md]: I started off trying to convert the names to all longer/more meaningful, but missed some towards the end. I will fix to use longer names throughout

**Comment by nevillelyh** [comment]
Another minor nit: GH CSS doesn't line wrap comments in code boxes. It'd be nice to manually wrap them so you don't have to scroll horizontally when reading.

**Pull request merged**